File: src/main/java/gg/mineral/practice/commands/tournament/JoinCommand.java
package gg.mineral.practice.commands.tournament;

import gg.mineral.practice.commands.PlayerCommand;
import gg.mineral.practice.entity.PlayerStatus;

import gg.mineral.practice.managers.EventManager;
import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.managers.TournamentManager;

import gg.mineral.practice.util.messages.impl.ErrorMessages;
import gg.mineral.practice.util.messages.impl.UsageMessages;
import lombok.val;

public class JoinCommand extends PlayerCommand {

    public JoinCommand() {
        super("join");
    }

    @Override
    public void execute(org.bukkit.entity.Player player, String[] args) {
        val profile = ProfileManager.getOrCreateProfile(player);

        if (args.length < 1) {
            profile.message(UsageMessages.JOIN);
            return;
        }

        if (profile.isInTournament()) {
            profile.message(ErrorMessages.ALREADY_IN_TOURNAMENT);
            return;
        }

        if (profile.isInEvent()) {
            profile.message(ErrorMessages.ALREADY_IN_EVENT);
            return;
        }

        if (profile.getPlayerStatus() != PlayerStatus.IDLE) {
            profile.message(ErrorMessages.YOU_ARE_NOT_IN_LOBBY);
            return;
        }

        val event = EventManager.getEventByName(args[0]);

        if (event != null)
            event.addPlayer(profile);

        val tournament = TournamentManager.getTournamentByName(args[0]);

        if (tournament != null)
            tournament.addPlayer(profile);

        if (event == null && tournament == null)
            profile.message(ErrorMessages.EVENT_TOURNAMENT_NOT_EXIST);
    }
}

File: src/main/kotlin/gg/mineral/practice/commands/tournament/JoinCommand.kt
package gg.mineral.practice.commands.tournament

import dev.rollczi.litecommands.annotations.argument.Arg
import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.PlayerStatus
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.managers.EventManager.getEventByName
import gg.mineral.practice.managers.TournamentManager
import gg.mineral.practice.util.messages.impl.ErrorMessages

@Command(name = "join")
class JoinCommand {
    @Execute
    fun execute(@Context profile: Profile, @Arg name: String) {
        profile.tournament?.let { return profile.message(ErrorMessages.ALREADY_IN_TOURNAMENT) }
        profile.event?.let { return profile.message(ErrorMessages.ALREADY_IN_EVENT) }
        if (profile.playerStatus !== PlayerStatus.IDLE) return profile.message(ErrorMessages.YOU_ARE_NOT_IN_LOBBY)

        val event = getEventByName(name)
        val tournament = TournamentManager.getTournament(name)

        if (event?.addPlayer(profile) == false && tournament?.addPlayer(profile) == false) profile.message(ErrorMessages.EVENT_TOURNAMENT_NOT_EXIST)
    }
}
